src/Inventory/Asset/NetworkPort.php

```diff
@@ -547,9 +592,9 @@ protected function portChanged(\stdClass $port, int $netports_id)
      * @param integer $items_id id of the item (0 if new)
      * @param string  $itemtype Item type
      * @param integer $rules_id Matched rule id, if any
-     * @param integer $ports_id Matched port id, if any
+     * @param array   $ports_id Matched port ids, if any
      */
-    public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = 0)
+    public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = [])
     {
```

Comment:
```suggestion
    public function rulepassed($items_id, $itemtype, $rules_id, $ports_id = [])
    {
        if (!is_array($ports_id)) {
           $ports_id = [$ports_id]; // Handle compatibility with previous signature.
        }
```

Reply:
I don't think we need to introduce a check for a legacy code, we have tests to avoid this. I've already fixed the signature everywhere i could find.

Reply:
This is a public method, so you may not be aware of some plugins that are using it.